Mathlib/Data/Rat/Cast/Order.lean

```diff
@@ -148,6 +149,97 @@ theorem preimage_cast_uIoc (a b : ℚ) : (↑) ⁻¹' uIoc (a : K) b = uIoc a b
 end LinearOrderedField
 end Rat
 
+namespace NNRat
+
+variable {K} [LinearOrderedSemifield K]
+
+theorem cast_strictMono : StrictMono ((↑) : ℚ≥0 → K) := fun m n h => by
+  rwa [NNRat.cast_def, NNRat.cast_def, div_lt_div_iff, ← Nat.cast_mul, ← Nat.cast_mul,
+    Nat.cast_lt (α := K), ← NNRat.lt_def]
+  · simp
+  · simp
+
+@[mono]
+theorem cast_mono : Monotone ((↑) : ℚ≥0 → K) :=
+  cast_strictMono.monotone
+
+/-- Coercion from `ℚ` as an order embedding. -/
+@[simps!]
+def castOrderEmbedding : ℚ≥0 ↪o K :=
+  OrderEmbedding.ofStrictMono (↑) cast_strictMono
+
+@[simp, norm_cast]
+theorem cast_le {m n : ℚ≥0} : (m : K) ≤ n ↔ m ≤ n :=
+  castOrderEmbedding.le_iff_le
+
+@[simp, norm_cast]
+theorem cast_lt {m n : ℚ≥0} : (m : K) < n ↔ m < n :=
+  cast_strictMono.lt_iff_lt
+
+@[simp]
+theorem cast_nonpos {n : ℚ≥0} : (n : K) ≤ 0 ↔ n ≤ 0 := by
+  norm_cast
+
+@[simp]
+theorem cast_pos {n : ℚ≥0} : (0 : K) < n ↔ 0 < n := by
+  norm_cast
+
+@[simp]
+theorem cast_lt_zero {n : ℚ≥0} : (n : K) < 0 ↔ n < 0 := by
+  norm_cast
+
+@[simp, norm_cast]
+theorem cast_min {a b : ℚ≥0} : (↑(min a b) : K) = min (a : K) (b : K) :=
```

Comment:
```suggestion
theorem cast_min (a b : ℚ≥0) : (↑(min a b) : K) = min (a : K) (b : K) :=
```
and below

Reply:
This is consistent with the Rat versions of the lemmas; I'd prefer to declare this out of scope too.

Reply:
I have opened #13644 to make your changes inconsistent with the Rat versions of the lemmas

Reply:
I'd appreciate it if you could push the corresponding changes to this PR once that lands; but it can wait till after your final exam!

Reply:
(or preferably, delegate this one first so that I can move on to the floorsemiring instance)

Reply:
Sure, I guess that's fixable later

maintainer merge